containerbuddy/config.go

```diff
@@ -311,8 +313,19 @@ func initializeConfig(config *Config) (*Config, error) {
 			return nil, ifaceErr
 		}
 
-		if service.ipAddress, err = GetIP(interfaces); err != nil {
-			return nil, err
+		if service.Endpoint == "" && service.ipAddress ==  "" {
+			if service.ipAddress, err = GetIP(interfaces); err != nil {
+				return nil, err
+			}
+		} else {
+			if ok := net.ParseIP(service.Endpoint); ok == nil {
+				if ip, err := net.LookupHost(service.Endpoint); err == nil {
```

Comment:
> If we're allowing hostnames at all, why not just pass them through as-is to Consul rather than trying to look them up here.

Good idea, I'll rewrite once I have a clue what to call the parameter.

>  you never assign the service address if it parses correctly as an IP

fixed.